migratedb.d/04.sql

```diff
@@ -673,13 +673,42 @@ BEGIN
     ALTER TABLE local_ega_download.requests DROP CONSTRAINT IF EXISTS requests_file_id_fkey;
     ALTER TABLE local_ega_download.requests ADD FOREIGN KEY(file_id) REFERENCES main_to_files(main_id) ON UPDATE cascade;
     ALTER TYPE local_ega_download.request_type ALTER ATTRIBUTE archive_file_checksum_type SET DATA TYPE sda.checksum_algorithm CASCADE;
+    ALTER TYPE local_ega_download.request_type ALTER ATTRIBUTE archive_type SET DATA TYPE local_ega.storage CASCADE;
 
     ALTER TABLE local_ega_ebi.fileindexfile DROP CONSTRAINT IF EXISTS fileindexfile_file_id_fkey;
     ALTER TABLE local_ega_ebi.fileindexfile ADD FOREIGN KEY(file_id) REFERENCES main_to_files(main_id) ON UPDATE cascade;
+    ALTER TABLE local_ega_ebi.fileindexfile ALTER COLUMN index_file_type TYPE local_ega.storage USING index_file_type::text::local_ega.storage;
+
+    DROP VIEW local_ega_ebi.file_index_file;
+    CREATE VIEW local_ega_ebi.file_index_file AS
+        SELECT m.stable_id AS file_id, index_file_id FROM local_ega_ebi.fileindexfile fif
+        INNER JOIN local_ega.main m ON fif.file_id=m.id;
 
     DROP VIEW IF EXISTS local_ega_ebi.file_dataset;
     DROP TABLE IF EXISTS local_ega_ebi.filedataset;
 
+    DROP VIEW local_ega_ebi.file;
+    CREATE VIEW local_ega_ebi.file AS
```

Comment:
does it make sense to create a new view based on `sda.files` instead of re-using the old naming scheme ?

or the plan is to drop `loca_ega_ebi` naming in another migration script and then create the views ?

Reply:
Well - it's mostly a preference thing :) I made this a view on the `local_ega` since that's what is was originally, and I felt that this maintained that reference. There's probably some imperceptible performance gain in having the view directly on `sda.files`, but it's unlikely to matter for us. Would you prefer this to be a view directly on `sda.files`?

Reply:
it is ok, where it is.